src/index.ts

```diff
@@ -1412,7 +1414,29 @@ export abstract class PinejsClientCore<PinejsClient> {
 
 		// Do not await this._request result, so that we can preserve
 		// the potentially enhanced promise-like result.
-		return this.callWithRetry(() => this._request(opts), retry);
+		return this.callWithRetry(async () => {
+			if (body) {
+				const blobKeys = Object.keys(body).filter(
+					(key) => body[key] instanceof Blob,
+				);
+				if (blobKeys.length > 0) {
+					opts.files = blobKeys.map(async (key) => {
+						const fileBlob = body[key] as Blob;
+						return {
+							fieldname: key,
+							mimetype: fileBlob.type,
+							buffer: {
+								type: 'Buffer',
+								data: await fileBlob.arrayBuffer(),
+							},
+							size: fileBlob.size,
+						};
+					});
+				}
+			}
```

Comment:
This should all be outside of the retry loop or otherwise this will be called on every single retry attempt which would be very wasteful

Reply:
Sure. I initially had it outside, moved it here because of `async..await` related access to `arrayBuffer()` data.